test/uk/gov/hmrc/excisemovementcontrolsystemapi/controllers/DraftExciseMovementControllerSpec.scala

```diff
@@ -70,85 +79,113 @@ class DraftExciseMovementControllerSpec
   "submit" should {
     "return 200" in {
       when(movementMessageService.saveNewMovement(any))
-        .thenReturn(Future.successful(Right(Movement("", "", None))))
+        .thenReturn(Future.successful(Right(Movement("123", "456", Some("789"), None, Instant.now))))
+
       val result = createWithSuccessfulAuth.submit(request)
 
       status(result) mustBe ACCEPTED
-    }
 
-    "call submission message service with the right eventId" in {
-      when(movementMessageService.saveNewMovement(any))
-        .thenReturn(Future.successful(Right(Movement("", "", None))))
-      await(createWithSuccessfulAuth.submit(request))
+      withClue("submit hte message") {
+        val captor = ArgCaptor[ValidatedXmlRequest[_]]
+        verify(submissionMessageService).submit(captor.capture)(any)
+        captor.value.message mustBe mockIeMessage
+      }
 
-      verify(submissionMessageService).submit(any)(any)
+      withClue("should get the box id") {
+        verify(notificationConnector).getBoxId(eqTo("clientId"))(any)
+      }
 
+      withClue("should save the new movement") {
+        val captor = ArgCaptor[Movement]
+        verify(movementMessageService).saveNewMovement(captor.capture)
+        captor.value.localReferenceNumber mustBe "123"
+        captor.value.consignorId mustBe "456"
+        captor.value.consigneeId mustBe Some("789")
+        captor.value.administrativeReferenceCode mustBe None
+        captor.value.messages mustBe Seq.empty
+      }
     }
 
     "call the add work item routine to create or update the database" in {
-
       when(movementMessageService.saveNewMovement(any))
         .thenReturn(Future.successful(Right(Movement("lrn", ern, None))))
 
       await(createWithSuccessfulAuth.submit(request))
 
       verify(workItemService).addWorkItemForErn("456", fastMode = true)
-
     }
 
-    "catch Future failure from Work Item service and log it but still process submission" in {
+    "return ACCEPTED if failing to add workItem " in {
       when(movementMessageService.saveNewMovement(any))
         .thenReturn(Future.successful(Right(Movement("lrn", ern, None))))
-
-      when(workItemService.addWorkItemForErn(any, any)).thenReturn(Future.failed(new MongoException("Oh no!")))
+      when(workItemService.addWorkItemForErn(any, any))
+        .thenReturn(Future.failed(new MongoException("Oh no!")))
 
       val result = createWithSuccessfulAuth.submit(request)
 
       status(result) mustBe ACCEPTED
-
-      verify(submissionMessageService).submit(any)(any)
     }
 
-    "return an error when EIS error" in {
-      when(submissionMessageService.submit(any)(any))
-        .thenReturn(Future.successful(Left(NotFound("not found"))))
+    "return an error" when {
+      "get box id return an error" in {
+        when(movementMessageService.saveNewMovement(any))
```

Comment:
Does this need to be mocked for this test? It shouldn't be called should it? (and next test)

Reply:
Yes you are right, this should not be called. Removed